lib/react_test_utils.dart

```diff
@@ -365,4 +365,40 @@ Element getDomNode(JsObject object) => object.callMethod('getDOMNode', []);
 JsObject mockComponent(JsObject componentClass, String mockTagName) {
   return _TestUtils.callMethod(
       'mockComponent', [componentClass, mockTagName]);
-}
\ No newline at end of file
+}
+
+/// Returns a ReactShallowRenderer instance
+///
+/// More info on using shallow rendering: https://facebook.github.io/react/docs/test-utils.html#shallow-rendering
+ReactShallowRenderer createRenderer() {
+  return new ReactShallowRenderer.jsObject(_TestUtils.callMethod('createRenderer', []));
+}
+
+/// ReactShallowRenderer wrapper
+///
+/// Usage:
+/// ```
+/// ReactShallowRenderer shallowRenderer = createRenderer();
+/// shallowRenderer.render(div({'className': 'active'}));
+///
+/// JsObject renderedOutput = shallowRenderer.getRenderOutput();
+/// expect(renderedOutput['props']['className'], 'active');
+/// ```
+///
+/// See react_with_addons.js#ReactShallowRenderer
+class ReactShallowRenderer {
+  final JsObject jsRenderer;
+
+  ReactShallowRenderer.jsObject(JsObject this.jsRenderer);
+
+  /// Get the rendered output. [render] must be called first
+  JsObject getRenderOutput() => jsRenderer.callMethod('getRenderOutput', []);
+
+  render(JsObject element, [Map context]) {
+    jsRenderer.callMethod('render', [element, context]);
```

Comment:
Ok, I think we're good. Somehow that cherry-pick went back, but I did them one-by-one this time and all seems well.